src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.cs

```diff
@@ -197,27 +199,32 @@ protected virtual void SetupNewTextView(IVsTextView textView)
                     v, Package.ComponentModel).AttachToVsTextView());
 
             var openDocument = wpfTextView.TextBuffer.AsTextContainer().GetRelatedDocuments().FirstOrDefault();
-            var isOpenMetadataAsSource = openDocument != null && openDocument.Project.Solution.Workspace.Kind == WorkspaceKind.MetadataAsSource;
 
             // If the file is metadata as source, and the user has the preference set to collapse them, then
             // always collapse all metadata as source
-            var collapseAllImplementations = isOpenMetadataAsSource &&
-                workspace.Options.GetOption<bool>(BlockStructureOptionsStorage.CollapseMetadataImplementationsWhenFirstOpened, openDocument.Project.Language);
+            var globalOptions = this.Package.ComponentModel.GetService<IGlobalOptionService>();
+            var masWorkspace = openDocument?.Project.Solution.Workspace as MetadataAsSourceWorkspace;
```

Comment:
❔ Why use the concrete type instead of the `Kind`?

Reply:
I need it to get the `FileService` from the workspace on like 209, so just seemed easier to do it all in one. Doesn't seem like there would be much to gain from the optimisation, but if you disagree i'm happy to change it.